internal/store/builder.go

```diff
@@ -134,23 +137,57 @@ func (b *Builder) WithVPAClient(c vpaclientset.Interface) {
 	b.vpaClient = c
 }
 
+// WithCustomResourceClients sets the customResourceClients property of a Builder.
+func (b *Builder) WithCustomResourceClients(cs map[string]interface{}) {
+	b.customResourceClients = cs
+}
+
+// WithUsingAPIServerCache configures whether using APIServer cache or not.
+func (b *Builder) WithUsingAPIServerCache(u bool) {
+	b.useAPIServerCache = u
+}
+
 // WithFamilyGeneratorFilter configures the family generator filter which decides which
 // metrics are to be exposed by the store build by the Builder.
 func (b *Builder) WithFamilyGeneratorFilter(l generator.FamilyGeneratorFilter) {
 	b.familyGeneratorFilter = l
 }
 
 // WithGenerateStoresFunc configures a custom generate store function
-func (b *Builder) WithGenerateStoresFunc(f ksmtypes.BuildStoresFunc, u bool) {
+func (b *Builder) WithGenerateStoresFunc(f ksmtypes.BuildStoresFunc) {
 	b.buildStoresFunc = f
-	b.useAPIServerCache = u
+}
+
+// WithGenerateCustomResourceStoresFunc configures a custom generate custom resource store function
+func (b *Builder) WithGenerateCustomResourceStoresFunc(f ksmtypes.BuildCustomResourceStoresFunc) {
+	b.buildCustomResourceStoresFunc = f
 }
 
 // DefaultGenerateStoresFunc returns default buildStores function
 func (b *Builder) DefaultGenerateStoresFunc() ksmtypes.BuildStoresFunc {
 	return b.buildStores
 }
 
+// DefaultGenerateCustomResourceStoresFunc returns default buildStores function
+func (b *Builder) DefaultGenerateCustomResourceStoresFunc() ksmtypes.BuildCustomResourceStoresFunc {
+	return b.buildCustomResourceStores
+}
+
+// WithCustomResourceStoreFactories returns configures a custom resource stores factory
+func (b *Builder) WithCustomResourceStoreFactories(fs ...customresource.RegistryFactory) {
```

Comment:
What about checking if `(customresource.RegistryFactory).Name()` overrides any of the already existing
stores inside of the pre-defined built-in list of stores (`store.availableStores`)?

In that light, what would happen if two `customresource.RegistryFactory` with the same name would be passed
in? Would this result in a silent, maybe difficult to debug situation?

I would plead for adding a check which returns an error if a store with the same name as the `customresource.ResourceFactory` already exists. With possibly a boolean somewhere (e.g. inside of `customresource.ResourceFactory`) to forcefully overwrite an existing store.

Reply:
Good idea. What about warning the overrides when the store already has an existed component with the same name? I'm afraid some guys may be not satisfied with built-in metrics and just want to make an override. We could just give a warning.

Reply:
I agree with the fact that some people might want to override built-in metrics. A simple warning to give the end-user an indication of what is happening seems like a good solution, much better than making the `customresource.ResourceFactory` more complex by adding an additional property.